bluesky/callbacks/broker.py

```diff
@@ -7,7 +7,34 @@
 import matplotlib.pyplot as plt
 
 
-class LiveImage(CallbackBase):
+class BrokerCallbackBase(CallbackBase):
+    """
+    Base class for callbacks which need filled documents
+
+    Parameters
+    ----------
+    fields: Iterable of str
+        Names of data field in an Event
+    db: Broker instance, optional
+        The Broker instance to pull the data from
+    """
+
+    def __init__(self, fields, *, db=None):
+        self.db = db
+        self.fields = fields
+
+    def event(self, doc):
+        for field in self.fields:
+            if doc.get('filled', {}).get(field):
+                pass
+            else:
+                if self.db is None:
+                    raise RuntimeError('Either the data must be pre-loaded or'
+                                       'a Broker instance must be provided.')
+                doc['data'][field] = self.db.fill_events([doc], field=field)
```

Comment:
`self.db.fill_events` is a generator. I think you need to do `doc['data'][field], = self.db.fill_events([doc], field=field)` (note trailing comma on left side). If I'm right, then we should be concerned that the test is passing now!